### DISCUSSION TITLE
How to Make TST Keep Focus on the Selected Child Tab When its Parent Tree is Collapsed

### DISCUSSION THREAD

**Discussion by Bokononist1917** [piroor/treestyletab#3414] [Q&A]
Currently, if you have a child tab selected and are viewing its webpage in the browser, and you collapse the parent tab that this child falls under in the TST sidebar, it changes tab focus from the previously selected child tab to the now collapsed parent tab, as if you had closed the child tab entirely. This behavior is nonsensical (given that collapsing a tab tree and closing a tab are two different actions) and to me, completely undesirable. Is there any way to force TST to keep the child tab selected and its website displayed regardless of whether you've collapsed the TST tree its a part of?

Note that I still want to preserve the behavior of focus jumping back to the parent when the child tab is closed entirely. I just hate the tab focus jumping all around whenever I collapse a tree of still-open tabs in order to make the sidebar cleaner and easier to navigate.

**Comment by piroor** [comment]
Does [TST Active Tab in Collapsed Tree](https://addons.mozilla.org/firefox/addon/tst-active-tab-in-collapsed-tr/) help you?

**Reply by Bokononist1917** [reply]
Not really. What I want is for the webpage of the initially selected child tab to remain displayed in the main browser window regardless of whether or not I collapse its parent-tree. The only time I ever want the actual currently-displayed-webpage to change is if I close a tab or manually select a different one. Tree collapse/expansion simply shouldn't affect it at all.

This just seems to give you a little preview icon of the last open tab before you collapsed the tree, without changing the behavior I don't like (forcing the browser to display the webpage of the parent tab when you collapse a tree which contains the tab your were initially viewing.)

**Reply by piroor** [reply]
> This just seems to give you a little preview icon of the last open tab before you collapsed the tree,

"TST Active Tab in Collapsed Tree" prevents TST to switch focus to the parent tab when a tree is collapsed. As the result you'll see collapsed tree in the sidebar and the browser window remains the contents of the last active child tab in the collapsed tree actually.
